### PR TITLE
Fix - When avatar's parent is deleted, avatar does not move

### PR DESCRIPTION
**Pull request by druiz17**
Fixed the issue of the when an avatar's parent is deleted the avatar is not able to move.

Link to issue - https://github.com/highfidelity/hifi/issues/9058

Test plan:
1. Enter in edit mode and create an object.
2. Run this script https://people.ucsc.edu/~druiz4/scripts/AvataeBug.js on the created entity, which makes the avatar the parent of the entity. 
3. One script is enabled move the entity around and your avatar will move with the entity.
4. Delete the entity,  your avatar should stay in pace and be able to move around.

QA Failed - If when performing step four, if the avatar ends up  suspended in the air or is unable to move.

### CODE REVIEW

File: libraries/entities/src/EntityTree.cpp [hunk 1]
```diff
@@ -447,6 +448,17 @@ void EntityTree::deleteEntity(const EntityItemID& entityID, bool force, bool ign
     _isDirty = true;
 }
 
+void EntityTree::checkEntity(const EntityItemID entityID) {
```
Comment (sethalves) [inline libraries/entities/src/EntityTree.cpp]: This looks good, but could I suggest some other name?  checkEntityForAvatarChildren or something?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4a1695ae20b44131bbd75f084b03426c78c66b08 available at: https://gist.github.com/02331f270547e53b47fd74e6eef463f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-4a1695a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-4a1695a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9c4ae9df5ad7d2e2ef2fd08dc2a016aa18bb49c4 available at: https://gist.github.com/02331f270547e53b47fd74e6eef463f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-9c4ae9d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-9c4ae9d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 66ff81f45066e2b890845ec3e8e0ee58fd38e5dc available at: https://gist.github.com/02331f270547e53b47fd74e6eef463f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-66ff81f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9087-66ff81f.dmg

**Comment by OmegaHeron** [comment]
QA Pass

Windows 10

This PR deleting entity with script leaves AV in place and able to move.  General smoke test passes as well.

Confirmed bad behavior on current master/rc.  Deleting test object places AV at origin and move locked.

**Pull request merged**